examples/cloud-datafusion-functions-plugins/README.md

```diff
@@ -0,0 +1,371 @@
+pso-cdf-plugins-287518
+
+**serviceFilePath:**  
+auto-detect
+
+**Screenshot:**
+
+![image](img/10-checkpoint_read_action_ui_pipeline_parameters.png)
+
+### TruncateTableAction
+
+**Label:**
+TruncateTableAction
+
+**Key path*:**  
+auto-detect
+
+**ProjectId* :**  
+pso-cdf-plugins-287518
+
+**Dataset** *
+bq_dataset
+
+**Table name***  
+bq_table_LOG
+
+![image](img/11-truncate_table_action_ui_pipeline_parameters.png)
+
+### Database source
+
+**Label** *
+Database
+
+**Reference Name***  
+test
+
+**Plugin Name***  
+sqlserver42
+
+**Plugin Type**  
+jdbc
+
+**Connection String**  
+jdbc:sqlserver://<fill in IP address of database server>:<db port>;databaseName=main;user=<fill in user>;password=<fill in password>;
+
+**Import Query:**  
+SELECT * FROM test WHERE last_update_datetime > '${latestWatermarkValue}'
+
+
+![image](img/12-database_source_ui_pipeline_parameters.png)
+
+
+### BigQuery sink
+
+**Label** *
+BigQuery
+
+**Reference Name***  
+bq_table_sink
+
+**Project ID**  
+pso-cdf-plugins-287518
+
+**Dataset***  
+bq_dataset
+
+**Table***  (write to a temporary table, e.g., bq_table_LOG)
+bq_table_LOG
+
+**Service Account File Path**  
+auto-detect
+
+**Schema**
+
+![image](img/13-bigquery_sink_ui_pipeline_parameters.png)
+
+### MergeLastUpdateTSAction
+
+**Label***  
+MergeLastUpdateTSAction
+
+**Key path***  
+auto-detect
+
+**Project ID***  
+pso-cdf-plugins-287518
+
+**Dataset name***  
+bq_dataset
+
+**Table name***  
+bq_table
+
+**Primary key list***  
+id
+
+**Update columns list***  
+id,name,last_update_datetime
+
+![image](img/14-merge_last_update_ts_action.png)
+
+### CheckPointUpdateAction
+
+**Label***  
+CheckPointUpdateAction
+
+**Specify the collection name in firestore DB***  
+PIPELINE_CHECKPOINTS
+
+**Specify the document name to read the checkpoint details***  
+INCREMENTAL_DEMO
+
+**Dataset name where incremental pull table exists***  
+bq_dataset
+
+**Table name that needs incremental pull***  
+bq_table
+
+**Specify the checkpoint column from incremental pull table***  
+last_update_datetime
+
+**serviceFilePath**  
+auto-detect
+
+**project**  
+pso-cdf-plugins-287518
+
+![image](img/15-checkpoint_update_action_ui_pipeline_parameters.png)
+
+## Building the CDF/CDAP Plugin .jar file and deploying into CDF/CDAP
+
+This plugin requires Java JDK1.8 and maven.
+
+1. To build the CDAP / CDF plugin jar, execute the following command on the root.
+```
+mvn clean compile package
+```
+
+2. You will find the generated .jar and .json file under target folder:
+   1. GoogleFunctions-1.6.jar
+   1. GoogleFunctions-1.6.json
+    
+1. Deploy GoogleFunctions-1.6.jar and GoogleFunctions-1.6.json into CDF/CDAP (note that if you have the same version already deployed then you’ll get an error that it already exists):
+    1. Delete GoogleFunctions artifact if the same version already exists.
+    1. Go to Control Center
+    1. Upload plugin by clicking on the circled green + button  
+    1. Pick the .jar / .json file 
```

Comment:
See: https://developers.google.com/style/filenames#file-type-names > Referring to file types

Reply:
thank you @mik-laj